project/npda/general_functions/rcpch_nhs_organisations.py

```diff
@@ -15,205 +15,29 @@
-        response.raise_for_status()
-
-        # Convert the response to choices list
-        organisation_list = [
-            (organisation.get("ods_code"), organisation.get("name"))
-            for organisation in response.json()
-        ]
-
-        return organisation_list
-    except HTTPError as e:
-        logger.error(f"HTTP error occurred: {e.response.text}")
-        return ERROR_RESPONSE
-    except Exception as e:
-        logger.error(f"An error occurred: {e}")
-        return ERROR_RESPONSE
-
-
-def get_all_nhs_organisations_affiliated_with_paediatric_diabetes_unit() -> (
-    List[Tuple[str, str]]
-):
-    """
-    This function returns all NHS organisations from the RCPCH dataset that are affiliated with a paediatric diabetes unit.
-    If an error occurs while fetching the data, it returns a list with a single tuple containing:
-        [("999", "An error occurred while fetching NHS organisations.")].
-
-    Returns:
-        List[Tuple[str, str]]: A list of tuples containing the ODS code and name of NHS organisations.
-    """
-    url = f"{settings.RCPCH_NHS_ORGANISATIONS_API_URL}/organisations/paediatric-diabetes-units"
-    ERROR_RESPONSE = [("999", "An error occurred while fetching NHS organisations.")]
-
-    try:
-        response = requests.get(url=url, timeout=10)  # times out after 10 seconds
-        response.raise_for_status()
-
-        # Convert the response to choices list
-        organisation_list = [
-            (organisation.get("ods_code"), organisation.get("name"))
-            for organisation in response.json()
-        ]
-
-        return organisation_list
-    except HTTPError as e:
-        logger.error(f"HTTP error occurred: {e.response.text}")
-        return ERROR_RESPONSE
-    except Exception as e:
-        logger.error(f"An error occurred: {e}")
-        return ERROR_RESPONSE
-
-
 def get_all_pz_codes_with_their_trust_and_primary_organisation() -> (
     List[Tuple[str, str]]
 ):
     """
     This function returns all NHS organisations from the RCPCH dataset that are affiliated with a paediatric diabetes unit.
-    Accepts a seed parameter - if True, it will seed the database with the data.
-    If False, it will return the data as a list of tuples.
-
-    If an error occurs while fetching the data, it returns a list with a single tuple containing:
-        [("999", "An error occurred while fetching NHS organisations.")].
 
     Returns:
         List[Tuple[str, str]]: A list of tuples containing the ODS code and name of NHS organisations.
     """
-    url = f"{settings.RCPCH_NHS_ORGANISATIONS_API_URL}/paediatric_diabetes_units/parent"
-    ERROR_RESPONSE = [("999", "An error occurred while fetching NHS organisations.")]
-
-    try:
-        response = requests.get(url=url, timeout=10)  # times out after 10 seconds
-        response.raise_for_status()
-
-        return response.json()
-
-    except HTTPError as e:
```

Comment:
This code was called from a seeding function via `manage.py` so I don't think there's much value in trapping errors, just let them bubble up and crash the command